components/core/src/clp/Array.hpp

```diff
@@ -0,0 +1,137 @@
+#ifndef CLP_ARRAY_HPP
+#define CLP_ARRAY_HPP
+
+#include <cstddef>
+#include <cstring>
+#include <memory>
+#include <string>
+#include <type_traits>
+#include <utility>
+
+#include "ErrorCode.hpp"
+#include "TraceableException.hpp"
+
+namespace clp {
+/**
+ * Class for a run-time fix-sized array.
+ * @tparam T The type of the element in the array. The type must be default constructable.
+ */
+template <typename T>
+requires(std::is_fundamental_v<T> || std::is_default_constructible_v<T>)
```

Comment:
```suggestion
requires(std::is_fundamental_v<T> || std::is_default_initializable<T>)
```

[This](https://stackoverflow.com/questions/62577956/what-is-the-difference-between-stddefault-initializable-and-stdis-default) seems to suggest `std::is_default_initializable` is stricter for our requirements?

Reply:
Make sense

Reply:
btw it's `[std::default_initializable](https://en.cppreference.com/w/cpp/concepts/default_initializable)`

Reply:
Right, my mistake. 5am, lol.